src/PhpSpec/CodeAnalysis/MagicAwareAccessInspector.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace PhpSpec\CodeAnalysis;
+
+use ReflectionMethod;
+use ReflectionProperty;
+
+final class MagicAwareAccessInspector implements AccessInspectorInterface
+{
+    /**
+     * @param object $object
+     * @param string $property
+     * @param bool $withValue
+     * @return bool
+     */
+    public function isPropertyAccessible($object, $property, $withValue = false)
+    {
+        if (!is_object($object)) {
```

Comment:
The fact it might not be an object is counterintuitive because the parameter is typehinted as an object, and stems from a bad design decision inside `Caller`

Let's keep the `is_object` check localised to `Caller` and not let it spill over to this new class.